meghanada.el

```diff
@@ -1411,6 +1411,26 @@ e.g. java.lang.annotation)."
                                    (format "\"%s\"" sym))))
     (message "client connection not established")))
 
+(defun meghanada--list-symbols ()
+  "List all available symbols."
+  (if (and meghanada--server-process (process-live-p meghanada--server-process))
+      (let* ((output (meghanada--send-request-sync "ls")))
+        (split-string output "\n"))
+       (message "client connection not established")))
+
+(defun meghanada-jump-symbol ()
+  "Jump to the specified symbol."
+  (interactive)
+  (if (and meghanada--server-process (process-live-p meghanada--server-process))
+      (let ((sym (ido-completing-read "Symbol: " (meghanada--list-symbols))))
```

Comment:
Please use `completing-read` as in the following code.

```
       (let ((sym (completing-read "Symbol: " (meghanada--list-symbols) nil nil)))
```

Reply:
That has way to strict filetring as it requires specifying the full package.

Maybe, it needs a custom predicates. Let me experiment.

Reply:
@iocanel I think that it is preferable to use `complting-read` because `helm` and `ido-completing- read-plus` and others can replace the backend of `complting-read` function.

Reply:
This makes sense. However, each of them has a little different way of filtering, so I think we need to be more explicit about it.

Reply:
@iocanel You should leave the completion function to a special package (`helm`, `ivy`, `ido-completing-read-plus`, etc). and it should be user-selectable. 
It adopts the same way in other packages.
(`lsp-mode`, etc)

I recommend using standard `completing-read`.